bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiAudioSink.java

```diff
@@ -81,28 +104,34 @@ public void process(AudioStream audioStream)
         } else {
             AudioFormat format = audioStream.getFormat();
             if (!AudioFormat.MP3.isCompatible(format) && !AudioFormat.WAV.isCompatible(format)) {
+                tryClose(audioStream);
                 throw new UnsupportedAudioFormatException("Currently only MP3 and WAV formats are supported.", format);
             }
 
-            if (audioStream instanceof URLAudioStream) {
-                // it is an external URL, the speaker can access it itself and play it
-                String url = ((URLAudioStream) audioStream).getURL();
-                logger.trace("Processing audioStream URL {} of format {}.", url, format);
-                handler.playURI(new StringType(url));
-            } else if (audioStream instanceof FixedLengthAudioStream) {
-                if (callbackUrl != null) {
-                    // we serve it on our own HTTP server for 30 seconds as Kodi requests the stream several times
-                    // Form the URL for streaming the notification from the OH2 web server
-                    String url = callbackUrl
-                            + audioHTTPServer.serve((FixedLengthAudioStream) audioStream, STREAM_TIMEOUT);
+            if (callbackUrl != null) {
+                // we serve it on our own HTTP server for 10 seconds as Kodi requests the stream several times
+                // Form the URL for streaming the notification from the OH web server
+                try {
+                    StreamServed streamServed = audioHTTPServer.serve(audioStream, STREAM_TIMEOUT, true);
+                    String url = callbackUrl + streamServed.url();
                     logger.trace("Processing audioStream URL {} of format {}.", url, format);
-                    handler.playNotificationSoundURI(new StringType(url));
-                } else {
-                    logger.warn("We do not have any callback url, so Kodi cannot play the audio stream!");
+                    handler.playNotificationSoundURI(new StringType(url), false);
+                } catch (IOException e) {
+                    tryClose(audioStream);
+                    logger.warn("Kodi binding was not able to handle the audio stream (cache on disk failed)", e);
```

Comment:
In all other audio sinks, it was decided to throw an exception (UnsupportedAudioStreamException) in this case.
Can you please do it also here for consistency ?

PS: Chromecast should also be updated).

Reply:
Indeed. Done